pkg/apis/core/validation/validation.go

```diff
@@ -1999,15 +1994,16 @@ func ValidatePersistentVolumeUpdate(newPv, oldPv *core.PersistentVolume, opts Pe
 	}
 
 	// PersistentVolumeSource should be immutable after creation.
-	if !apiequality.Semantic.DeepEqual(newPv.Spec.PersistentVolumeSource, oldPv.Spec.PersistentVolumeSource) {
-		pvcSourceDiff := cmp.Diff(oldPv.Spec.PersistentVolumeSource, newPv.Spec.PersistentVolumeSource)
-		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "persistentvolumesource"), fmt.Sprintf("spec.persistentvolumesource is immutable after creation\n%v", pvcSourceDiff)))
-	}
-	allErrs = append(allErrs, ValidateImmutableField(newPv.Spec.VolumeMode, oldPv.Spec.VolumeMode, field.NewPath("volumeMode"))...)
+	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(
+		newPv.Spec.PersistentVolumeSource,
+		oldPv.Spec.PersistentVolumeSource,
+		field.NewPath("spec", "persistentvolumesource"))...,
+	)
+	allErrs = append(allErrs, apimachineryvalidation.ValidateImmutableField(newPv.Spec.VolumeMode, oldPv.Spec.VolumeMode, field.NewPath("volumeMode"))...)
```

Comment:
The path is incorrect, it was incorrect before, it's still incorrect, shouldn't it be `field.NewPath("spec", "volumeMode")`?

Reply:
Indeed, I have just changed it

Reply:
Indeed, just changed it